plugins/plugin-chart-echarts/src/BoxPlot/types.ts

```diff
@@ -0,0 +1,35 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { QueryFormData } from '@superset-ui/core';
+
+export type BoxPlotQueryFormData = QueryFormData & {
+  numberFormat?: string;
+  whiskerOptions?: BoxPlotFormDataWhiskerOptions;
+  xTickLayout?: BoxPlotFormXTickLayout;
+};
+
+export type BoxPlotFormDataWhiskerOptions =
+  | 'Tukey'
+  | 'Min/max (no outliers)'
+  | '2/98 percentiles'
+  | '9/91 percentiles';
+
+export type BoxPlotFormXTickLayout = '45°' | '90°' | 'auto' | 'flat' | 'staggered';
```

Comment:
90° rotation added. Staggered isn't really supported in ECharts BoxPlot (neither was it in the previous plugin), but leaving this here for backward compatiblity + if it gets implemented in ECharts sometime in the future.